pkg/fakes/fake_network_store.go

```diff
@@ -97,23 +134,24 @@ func (f *FakeNetworkStore) GetNetworks(plainFilters filters.Args) ([]dockerTypes
 
 	for _, key := range f.SortedIDs() {
 		network := f.networks[key]
-		// if we're filtering on stack ID, and this network doesn't match, then
-		// we should skip this network
+
+		// if we're filtering on stack ID, and this network doesn't
+		// match, then we should skip this network
 		if hasFilter && network.Labels[types.StackLabel] != stackID {
 			continue
 		}
 		// otherwise, we should append this network to the set
-		if err := f.causeAnError(nil, "GetNetworks", *network); err != nil {
+		if err := f.maybeTriggerAnError("GetNetworks", *network); err != nil {
 			return nil, err
 		}
-		networks = append(networks, *network)
+		networks = append(networks, *CopyNetworkResource(*network))
```

Comment:
I'm seeing, in this change set, a fair bit of derefing pointers. Just to make sure, you're aware that if the struct being derefed itself contains more pointers, those pointers will still point to the originals? I'm pretty sure you already know this, but I wanted to check before I merge.